addons/account/models/account_move.py

```diff
@@ -740,22 +742,21 @@ def check_full_reconcile(self):
         In case of full reconciliation, all moves belonging to the reconciliation will belong to the same account_full_reconcile object.
         """
         # Get first all aml involved
-        part_recs = self.env['account.partial.reconcile'].search(['|', ('debit_move_id', 'in', self.ids), ('credit_move_id', 'in', self.ids)])
-        amls = self
-        todo = set(part_recs)
-        seen = set()
+        todo = self.env['account.partial.reconcile'].search_read(['|', ('debit_move_id', 'in', self.ids), ('credit_move_id', 'in', self.ids)], ['debit_move_id', 'credit_move_id'])
+        amls = set(self.ids)
+        seen = self.env['account.partial.reconcile']
```

Comment:
reading the code, it seems `seen` could be changed into a set of ids, making the browse of line 751 useless